### ISSUE TITLE
How use contentFor in engines

### ISSUE THREAD

**Issue by brunowego** [ember-engines/ember-engines#409]
So, I have a engine with `lazyLoading: false` inside a fresh ember app. My engine have addons and one of them have the code below:

```js
# index.js
/* eslint-env node */
'use strict';

module.exports = {
  name: 'ember-cli-neutral',

  contentFor(type, config) {
    if (type === 'head') {
      if (config.googleFonts) {
        let families = config.googleFonts.join('|');

        return '<link href="https://fonts.googleapis.com/css?family=' + families + '" rel="stylesheet">'; // eslint-disable-line
      }
    }
  }
};
```

```js
# environment.js
/* eslint-env node */
'use strict';

module.exports = () => {
  const ENV = {
    googleFonts: [
      'Roboto:400,500'
    ]
  };

  return ENV;
};
```

If I run engine separately the link of google fonts works perfectly, but if I run inside the fresh ember app not.

Perhaps a feature for future or it's outside the scope of this project?

**Comment by crodriguez1a** [comment]
I also came across this. Not sure if it was intended. I'd be happy to try and debug if it's not.

**Comment by rwjblue** [comment]
The fix is to update the `contentFor` in `ember-engines`'s `lib/engine-addon.js` [here](https://github.com/ember-engines/ember-engines/blob/master/lib/engine-addon.js#L873) to invoke `contentFor` on all of its own `addons`.

**Comment by rwjblue** [comment]
I'm not 100% sure that we want this though, it seems somewhat odd to allow other addons to mutate the main apps `index.html` (considering that engines are essentially an isolation primitive)?

@trentmwillis - Thoughts?

**Comment by trentmwillis** [comment]
> it seems somewhat odd to allow other addons to mutate the main apps index.html (considering that engines are essentially an isolation primitive)

Agreed. However, it also seems that we would want Engines to be able to use addons that have `contentFor` hooks. I'm not sure what the best solution here, but I lean towards enabling `contentFor` for an Engine's addons for consistency, as I believe it is what users would expect.

**Comment by denzo** [comment]
Just checking if there were any discussions around that.

We are using `ember-basic-dropdown` inside an engine and because it relies on the [`contentFor` hook](https://github.com/cibernox/ember-basic-dropdown/blob/master/index.js#L25) it does not work.

@cibernox @trentmwillis @rwjblue

**Comment by sumeetattree** [comment]
Came across this issue while splitting our large app into smaller engines. Currently I am working around it by including the addon inside the main app as well. It's not ideal, but does the job. Are we going to see this change land in the near future?

> I'm not 100% sure that we want this though, it seems somewhat odd to allow other addons to mutate the main apps `index.html` (considering that engines are essentially an isolation primitive)?
> 
> @trentmwillis - Thoughts?

@rwjblue IMO this doesn't seem to be that big an issue. The addons are already breaking the isolation premise by directly modifying the app's `index.html`. It wouldn't be that much of a stretch to allow engines to modify the `index.html` as well.

**Comment by ascudder** [comment]
I wanted to add my support for allowing engines to use `contentFor` as well. One of our engines needs to pull in a 3rd party dependency by adding to the index.hrml file. My first thought was that "engines are just a kind of addon," so I expected it would work, but was surprised to find out that it doesn't work and to find this issue.
